src/NServiceBus.Core/Hosting/HostingComponent.cs

```diff
@@ -0,0 +1,134 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Net;
+    using System.Runtime;
+    using Hosting;
+    using Settings;
+    using Support;
+
+    class HostingComponent
+    {
+        HostingComponent(HostInformation hostInformation)
+        {
+            HostInformation = hostInformation;
+        }
+
+        public static HostingComponent Initialize(Configuration configuration,
+            ContainerComponent containerComponent,
+            PipelineComponent pipelineComponent,
+            string endpointName,
+            ReadOnlySettings settings)
+        {
+            var hostInformation = new HostInformation(configuration.HostId, configuration.DisplayName, configuration.Properties);
+
+            //for backwards compatibility, can be removed in v8
+            containerComponent.ContainerConfiguration.ConfigureComponent(() => hostInformation, DependencyLifecycle.SingleInstance);
+
+            pipelineComponent.PipelineSettings.Register("AuditHostInformation", new AuditHostInformationBehavior(hostInformation, endpointName), "Adds audit host information");
```

Comment:
didn't we conclude that this type of information should be added by the "send/outgoing component"? To me it looks like a smell that the host depends on the pipeline. Host & container seem to be very base level components which shouldn't be aware of the inner works of the endpoint?

Reply:
Yes, we'll move those 2 once you have the component added?

Reply:
ok, we just need to make sure we won't forget this :D